unit/java_bytecode/generate_java_generic_type.cpp

```diff
@@ -0,0 +1,48 @@
+/*******************************************************************\
+
+ Module: Unit tests for generating new type with generic parameters
+         substitued for concrete types
+
+ Author: DiffBlue Limited. All rights reserved.
+
+\*******************************************************************/
+
+#include <catch.hpp>
+
+#include <util/symbol_table.h>
+
+#include <java_bytecode/generate_java_generic_type.h>
+#include <java_bytecode/java_types.h>
+
+#include <iostream>
+
+SCENARIO(
+  "generate_java_generic_type",
+  "[CORE][java_bytecode][generate_java_generic_type")
+{
+  GIVEN("A generic java type and a concrete substitution")
```

Comment:
Add more unit tests for classes with multiple fields, multiple generic parameters, nested generic parameters.